### PR TITLE
 Private Events

### PR DESCRIPTION
**Pull request by Shaher-11**
In the private events project the user can create events and the user can attend many events. An event can be attended by many users. This required us to model many-to-many relationships and also to be very conscious about our foreign keys.
The project emphasizing the core functionalities rather than the UI.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <% if user_signed_in? %>
+
+    <h4>Created events - <%= @created_events.count %></h4>
+    <% @created_events.each do |event| %>
+          <h2><%= event.title %></h2>
+          <%= display_name(event) %>
+          <p><%= truncate(event.description,length:20) %></p>
+          <%= link_to 'Show more', event_path(event) %>
+          Created at <%= (event.created_at) %> 
+    <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb]: Please do not use conditions here. It's app logic so move it to view helpers.

File: app/views/events/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(event.errors.count, "error") %> prohibited this event from being saved:</h2>
+
+      <ul>
+        <% event.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/events/_form.html.erb:12]: Please do not use conditions here. It's app logic so move it to view helpers.

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <h5><%= @event.location %></h5>
+
+        <h2>Description</h2>
+        <p><%= @event.description %></p>
+
+      <%= show_registration(@event) %>
+
+ <% if current_user == @event.creator %>
+      <%= link_to 'Delete', event_path, method: :delete, data: { confirm: "Delete this event?"} %>
+    <% end %>
```
Comment (SajjadAhmad14) [inline app/views/events/show.html.erb:24]: Please do not use conditions here. It's app logic so move it to view helpers.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] You have a professional repo and it's very easy to follow your README to clone the project locally.
- [x] Thank you for the extra work you have put in by using devise gem.
- [x] It looks awesome when you show flash errors in red color.

## To Improve 🙏
- [x] Your repo does not have a master branch that is used for production. Kindly add it! 
- [x] The user's show page should display events that the user is attending to separated in the past and upcoming. You can create variables @past_events and @upcoming_events by using the event model scope. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png).
- [x] Some validations and associations tests are failing. KIndly fix them all!
 - [x] You should write integration tests for authentication and event management features. Please see the [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png).
- [x] Views should have no application logic. You have some logic in events and users views. See inline comments for exact lines of code. You should move app logic to View helpers. Please do go through these resources:
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers — Best Practices Guide](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)

# Optional

- [x]  It's always secure to configure only those routes which are used. Events is only using :new, :create, :show, and :index routes.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by Shaher-11** [comment]
Done all the requested changes

**Review by Temmarie — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi @Shaher-11 and @nikoescobal  ! 
### You have done well so far :clap: However, you did not finish all the previous review changes. Kindly work on them and submit them for another code review.

- [x] You still have some tests that are failing. Kindly fix them.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Comment by Shaher-11** [comment]
Fix the test and made sure they all pass and run the tests one by one
-rspec spec/event_spec.rb
- rspec spec/authentication_spec.rb
-rspec spec/models/event_spec.rb
-rspec spec/models/event_user_spec.rb
-rspec spec/models/user_spec.rb

**Review by whiz25 — APPROVED** [review]
## Approved ✅ 
Nice work making the requested changes. All test cases now pass.
Merge and proceed to the next project.


### Martin Afani
#### Slack: whiz25

**Pull request merged**